lib/ex_aws/ecs.ex

```diff
@@ -0,0 +1,1091 @@
+  def describe_task_definition(task_definition) do
+    request(:describe_task_definition, %{"taskDefinition" => task_definition})
+  end
+
+  @doc """
+  Describes a specified task or tasks in the `default` cluster.
+
+  See also `describe_tasks/2`
+  """
+  @spec describe_tasks(tasks :: list(binary)) :: ExAws.Operation.JSON.t
+  def describe_tasks(tasks) do
+    request(:describe_tasks, %{"tasks" => tasks})
+  end
+
+  @doc """
+  Describes a specified task or tasks in a cluster.
+  """
+
+  @spec describe_tasks(tasks :: list(binary), cluster :: binary) :: ExAws.Operation.JSON.t
+  def describe_tasks(tasks, cluster) do
+    request(:describe_tasks, %{"tasks" => tasks, "cluster" => cluster})
+  end
+
+  @type discover_poll_endpoint_opts :: [
+    {:cluster, binary} |
+    {:container_instance, binary}
+  ]
+  @doc """
+  Returns an endpoint for the Amazon EC2 Container Service agent to
+  poll for updates.
+
+  _This action is only used by the Amazon EC2 Container Service agent,
+  and it is not intended for use outside of the agent_
+  """
+  @spec discover_poll_endpoint() :: ExAws.Operation.JSON.t
+  @spec discover_poll_endpoint(opts :: discover_poll_endpoint_opts) :: ExAws.Operation.JSON.t
+  def discover_poll_endpoint(opts \\ []) do
+    data = opts
+    |> normalize_opts
+
+    request(:discover_poll_endpoint_opts, data)
+  end
+
+  @type list_attribute_opts :: [
+    {:attribute_name, binary} |
+    {:attribute_value, binary} |
+    {:cluster, binary} |
+    {:max_results, 1..100} |
+    {:next_token, binary}
+  ]
+
+  @doc """
+  Lists the attributes for Amazon ECS resources within a specified target type and cluster.
+  When you specify a target type and cluster, `list_attributes` returns a list of attribute objects,
+  one for each attribute on each resource. You can filter the list of results to a single
+  attribute name to only return results that have that name.
+  You can also filter the results by attribute name and value, for example,
+  to see which container instances in a cluster are running a Linux AMI (`ecs.os-type=linux`).
+  """
+  @spec list_attributes(target_type :: attribute_target_type) :: ExAws.Operation.JSON.t
+  @spec list_attributes(target_type :: attribute_target_type, opts :: list_attribute_opts) :: ExAws.Operation.JSON.t
+  def list_attributes(target_type, opts \\ []) do
+    data = opts
+    |> normalize_opts
+    |> Map.merge(%{"targetType" => to_string(target_type)})
+    request(:list_attributes, data)
+  end
+
+  @type list_clusters_opts :: [
+    {:max_results, 1..100} |
+    {:next_token, binary}
+  ]
+
+  @doc """
+  Returns a list of existing clusters.
+  """
+  @spec list_clusters() :: ExAws.Operation.JSON.t
+  @spec list_clusters(opts :: list_clusters_opts) :: ExAws.Operation.JSON.t
+  def list_clusters(opts \\ []) do
+    data = opts
+    |> normalize_opts
+    request(:list_clusters, data)
+  end
+
+  @type container_instance_status :: [
+    :ACTIVE |
+    :DRAINING
+  ]
```

Comment:
If we're going to typespec these I think it would be more idiomatic to have them lower case. This does require that we handle upcasing them.